src/test/rpc/AccountTx_test.cpp

```diff
@@ -28,6 +28,78 @@ namespace test {
 
 class AccountTx_test : public beast::unit_test::suite
 {
+    // A data structure used to describe the basic structure of a 
+    // transactions array node as returned by the account_tx RPC command.
+    struct NodeSanity
+    {
+        int const index;
+        Json::StaticString const& txType;
+        std::initializer_list<char const*> created;
+        std::initializer_list<char const*> deleted;
+        std::initializer_list<char const*> modified;
+    };
+
+    // A helper method tests can use to validate returned JSON vs NodeSanity.
+    void
+    checkSanity (Json::Value const& txNode, NodeSanity const& sane)
+    {
+        BEAST_EXPECT(txNode[jss::validated].asBool() == true);
+        BEAST_EXPECT(
+            txNode[jss::tx][sfTransactionType.jsonName].asString() == 
+            sane.txType);
+
+        // Make sure all of the expected node types are present.
+        std::vector<std::string> createdNodes {};
+        std::vector<std::string> deletedNodes {};
+        std::vector<std::string> modifiedNodes{};
```

Comment:
I'd replace the `std::vector` here and the `std::initializer_list` above with a boost `flat_set`

Using unsorted vectors and initializer_list complicates the test:
1) They require a lambea to sort the values before they are compared
2) One of the parameters to the lambda `cmpNodeTypes` is modified, something I
wouldn't expect from what looks like a pure function
3) The other parameter is copied

Replaceing these containers with a `flat_set` gets rid of the lambda, and lets
us use the clearer form: `BEAST_EXPECT(createdNodes == sane.created)`
 Here's a patch to cherry pick/review if you want (that patch also fixes some white space issues in this file)

Patch: https://github.com/seelabs/rippled/commit/bcda862d76498b9ddb78fe1b6a6653a70ce2e8c3